finp2p-contracts/contracts/token/ERC20/FINP2POperatorERC20.sol

```diff
@@ -174,11 +174,27 @@ contract FINP2POperatorERC20 is IFinP2PAsset, IFinP2PEscrow, AccessControl {
 
         Asset memory asset = assets[assetId];
         uint256 balance = IERC20(asset.tokenAddress).balanceOf(issuer);
-        require(balance >= quantity, "Not sufficient balance to redeem");
-
-        ERC20WithOperator(asset.tokenAddress).burnFrom(issuer, quantity);
-
-        emit Redeem(assetId, account, quantity);
+        if (operationId == 0x00000000000000000000000000000000) {
+           require(balance >= quantity, "Not sufficient balance to redeem");
+        }
+        if (operationId != 0x00000000000000000000000000000000) {
+                require(haveContract(operationId), "Contract does not exists");
+
+                Lock storage lock = locks[operationId];
+                require(quantity == lock.amount, "Amount to redeem is not equal to locked amount for this operationId");
+        }
+
+        if (operationId != 0x00000000000000000000000000000000) {
+                ERC20WithOperator(asset.tokenAddress).burnFrom(address(this), quantity);
```

Comment:
Please add small comment into a contract how do we pick which account to withdraw from

Reply:
done